llvm/lib/Target/Mips/MipsScheduleI6400.td

```diff
@@ -0,0 +1,447 @@
+//==- MipsScheduleI6400.td - I6400 Scheduling Definitions --*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+def MipsI6400Model : SchedMachineModel {
+  int IssueWidth = 2;         // 2x dispatched per cycle
+  int MicroOpBufferSize = 0;
+  int LoadLatency = 3;
+  int MispredictPenalty = 8;
+
+  let CompleteModel = 1;
+  let FullInstRWOverlapCheck = 1;
+
+  list<Predicate> UnsupportedFeatures = [HasMips64r5, HasMips32r5,
+                                         InMicroMips, InMips16Mode, HasCnMips,
+					 HasCnMipsP,  HasDSP, HasDSPR2,
+					 HasMips3D, HasMT, HasCRC,
+					 NotMips32r6, NotMips64r6, HasEVA];
+}
+
+let SchedModel = MipsI6400Model in {
+
+  // AGEN Pipelines
+  // ==============
+  def I6400AGEN : ProcResource<1> { let BufferSize = 0; }
```

Comment:
Setting `BufferSize = 0` on only some of the pipes is not enough I'm afraid. If this is an in-order core then every pipes should have `BufferSize = 0`. You can do it by
```
let BufferSize = 0 {
  def PipeA : ProcResource<1>;
  def PipeB : ProcResource<1>;
  ...
}
```